abstractions/python/kiota/abstractions/serialization/parse_node.py

```diff
@@ -0,0 +1,160 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from datetime import datetime
+from enum import Enum
+from io import BytesIO
+from typing import Callable, List, Optional, TypeVar
+from uuid import UUID
+
+from .parsable import Parsable
+
+T = TypeVar("T")
+
+U = TypeVar("U", bound=Parsable)
+
+K = TypeVar("K", bound=Enum)
+
+
+class ParseNode(ABC):
+    """
+    Interface for a deserialization node in a parse tree. This interace provides an abstraction
+    layer over serialization formats, libraries and implementations.
+    """
+    @abstractmethod
+    def get_string_value(self) -> str:
+        """Gets the string value of the node
+
+        Returns:
+            str: The string value of the node
+        """
+        pass
+
+    @abstractmethod
+    def get_child_node(self, identifier: str) -> ParseNode:
+        """Gets a new parse node for the given identifier
+
+        Args:
+            identifier (str): The identifier of the current node property
+
+        Returns:
+            ParseNode: A new parse node for the given identifier
+        """
+        pass
+
+    @abstractmethod
+    def get_boolean_value(self) -> Optional[bool]:
+        """Gets the boolean value of the node
+
+        Returns:
+            bool: The boolean value of the node
+        """
+        pass
+
+    @abstractmethod
+    def get_int_value(self) -> Optional[int]:
+        """Gets the integer value of the node
+
+        Returns:
+            int: The integer value of the node
+        """
+        pass
+
+    @abstractmethod
+    def get_float_value(self) -> Optional[float]:
+        """Gets the float value of the node
+
+        Returns:
+            float: The integer value of the node
+        """
+        pass
+
+    @abstractmethod
+    def get_uuid_value(self) -> Optional[UUID]:
+        """Gets the UUID value of the node
+
+        Returns:
+            UUID: The GUID value of the node
+        """
+        pass
+
+    @abstractmethod
+    def get_datetime_offset_value(self) -> Optional[datetime]:
```

Comment:
see the additional types we've introduced in #1041 and #1017

Reply:
Added in https://github.com/microsoft/kiota/pull/1058/commits/2b2dcec31d3448b6f1d9fda5b8a989675806916d